crates/amaru/tests/evaluate_ledger_states.rs

```diff
@@ -0,0 +1,274 @@
+    let _pulsing_rew = d.skip()?;
+    let _pool_distr = d.skip()?;
+    let _stashed = d.skip()?;
+
+    Ok((
+        DefaultValidationContext::new(utxos_map),
+        current_pparams_hash,
+        GovernanceActivity {
+            consecutive_dormant_epochs: u64::from(number_of_dormant_epochs) as u32,
+        }
+    ))
+}
+
+fn decode_segregated_parameters(
+    dir: &PathBuf,
+    hash: &minicbor::bytes::ByteSlice,
+) -> Result<ProtocolParameters, Box<dyn std::error::Error>> {
+    let pparams_file_path = fs::read_dir(dir)?
+        .filter_map(|entry| entry.ok().map(|e| e.path()))
+        .find(|path| {
+            path.file_name()
+                .map(|filename| filename.to_str() == Some(&hex::encode(hash.as_ref())))
+                .unwrap_or(false)
+        })
+        .ok_or("Missing pparams file")?;
+
+    let pparams_file = fs::read(pparams_file_path)?;
+
+    let pparams = cbor::Decoder::new(&pparams_file).decode()?;
+
+    Ok(pparams)
+}
+
+fn import_vector(
+    record: TestVector,
+    ledger_dir: &PathBuf,
+    era_history: &EraHistory,
+    pparams_dir: &PathBuf,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let epoch = peek_epoch(&record.initial_state)?;
+    let point = Point::Specific(
+        epoch * 86400,
+        hex::decode("0000000000000000000000000000000000000000000000000000000000000000")?
+    );
+
+    let mut decoder = minicbor::Decoder::new(&record.initial_state);
+    let (mut validation_context, pparams_hash, governance_activity) = decode_ledger_state(&mut decoder)?;
+
+    let protocol_parameters = decode_segregated_parameters(pparams_dir, &pparams_hash)?;
+
+    for (ix, event) in record.events.into_iter().enumerate() {
+        let (tx_bytes, success, slot): (Bytes, bool, u64) = match event {
+            TestVectorEvent::Transaction(tx, success, slot) => (tx, success, slot),
+            _ => continue,
+        };
+        let tx: MintedTx<'_> = minicbor::decode(&*tx_bytes)?;
+
+        let tx_body: KeepRaw<'_, MintedTransactionBody<'_>> = tx.transaction_body.clone();
+        let tx_witness_set: MintedWitnessSet<'_> = tx.transaction_witness_set.deref().clone();
+        let tx_auxiliary_data =
+            Into::<Option<KeepRaw<'_, AuxiliaryData>>>::into(tx.auxiliary_data.clone())
+                .map(|aux_data| Hasher::<256>::hash(aux_data.raw_cbor()));
+
+        let pointer = TransactionPointer {
+            slot: point.slot_or_default(),
+            transaction_index: ix,
+        };
+
+        // Run the transaction against the imported ledger state
+        let result = transaction::execute(
+            &mut validation_context,
+            &Network::Testnet,
+            &protocol_parameters,
+            era_history,
+            &governance_activity,
+            pointer,
+            true,
+            tx_body,
+            &tx_witness_set,
+            tx_auxiliary_data,
+        );
+
+        match result {
+            Ok(()) => {
+                if success {
```

Comment:
Again, this seems to be _acceptable_ for a first iteration of it, but it's important we are actually comparing the final state instead of just checking for the absence of an error.

A transaction could be executed without an error but provide the incorrect state transitions for the ledger, and that's what these tests should be validating for us.